webapp/src/index.tsx

```diff
@@ -581,8 +592,17 @@ export default class Plugin {
                                 startAt: data[i].call?.start_at,
                                 ownerID: data[i].call?.owner_id,
                                 hostID: data[i].call?.host_id,
+                                dismissedNotification: data[i].call?.dismissed_notification || [],
                             },
                         });
+
+                        const callExists = incomingCalls(store.getState()).findIndex((ic) => ic.callID === data[i].channel_id) >= 0;
+                        if (data[i].call && !callExists) {
+                            // This allows us time to load channels and call states
+                            setTimeout(() => {
+                                store.dispatch(incomingCallOnChannel(data[i].channel_id, data[i].call.host_id, data[i].call.start_at));
+                            }, 2000);
+                        }
```

Comment:
Not in love with this :p Could you help me understand the problem/requirements?

Reply:
Thanks for pointing this out -- I had fixed the underlying problem and forgot to remove it. (I turned the incomingCallChannel into a thunk that would load the channel and the host if they weren't in the store, and that's what the delay was originally trying to fix.)